addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,190 @@
+
+    def _l10n_ro_edi_get_attachment_file_name(self):
+        """ Returns the signature file attachment's name from `l10n_ro_edi.document`/`invoice_sent` """
+        self.ensure_one()
+        return f"ciusro_{self.name.replace('/', '_')}.xml"
+
+    def _l10n_ro_edi_get_failed_documents(self):
+        """ Shorthand for getting all l10n_ro_edi.document in invoice_sending_failed state """
+        return self.l10n_ro_edi_document_ids.filtered(lambda d: d.state == 'invoice_sending_failed')
+
+    def _l10n_ro_edi_get_sending_and_failed_documents(self):
+        """ Shorthand for getting all l10n_ro_edi.document in invoice_sending and invoice_sending_failed state """
+        return self.l10n_ro_edi_document_ids.filtered(lambda d: d.state in ('invoice_sending', 'invoice_sending_failed'))
+
+    ################################################################################
+    # Send Logics
+    ################################################################################
+
+    def _l10n_ro_edi_get_pre_send_errors(self, xml_data):
+        """ Compute all possible common errors before sending the XML to the SPV """
+        self.ensure_one()
+        errors = []
+        if not self.company_id.l10n_ro_edi_access_token:
+            errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
+        if not xml_data:
+            errors.append(_('CIUS-RO XML attachment not found.'))
+        return errors
+
+    def _l10n_ro_edi_send_invoice(self, xml_data):
+        """ This method send xml_data to the Romanian SPV using the single invoice's (self) data.
+            The invoice's company and move_type will be used to calculate the required params in the send request.
+            The state of the document deletion/creation are as follows:
+             - Pre-check any errors from invoice's company before sending
+                - if error -> delete all error documents, create a new error document
+                - else -> continue to the next step
+             - Send to E-Factura, and based on the result:
+                - if error -> delete all error documents, create a new error document
+                - if success -> delete all error & sending documents, create a new sending document
+            :param xml_data: string of the xml data to be sent
+        """
+        self.ensure_one()
+        if errors := self.company_id._l10n_ro_edi_get_errors_pre_request():
+            self._l10n_ro_edi_get_failed_documents().unlink()
+            self._l10n_ro_edi_create_document_invoice_sending_failed(self, '\n'.join(errors))
+            return
+
+        self.env['res.company']._with_locked_records(self)
+        result = self.env['l10n_ro_edi.document']._request_ciusro_send_invoice(
+            company=self.company_id,
+            xml_data=xml_data,
+            move_type=self.move_type,
+        )
+        if 'error' in result:
+            self._l10n_ro_edi_get_failed_documents().unlink()
```

Comment:
it seems you are removing the failed documents twice (see line 129)

Reply:
The first one have an early return after removing so it's fine.